File: 13-roman-to-integer/13-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman={"I":1,
             "V":5,
             "X":10,
             "L":50,
             "C":100,
             "D":500,
             "M":1000,
             "IV":4,
             "IX":9,
             "XL":40,
             "XC":90,
             "CD":400,
             "CM":900
        }
        
        num=0
        lst = 0
        for i in s[::-1]:
            n = roman.get(i)
            if  n >= lst:
                num += n
            
            else:
                num -= n
            lst = n
            
        return num
            
            
            
            

        return num    
            

File: 13-roman-to-integer/13-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        
char [] a =  s.toCharArray();
int  prev = 0;
int RunningSum =0;
        
  Map<Character, Integer> lookup = new HashMap<Character, Integer>();
lookup.put('I', 1);
lookup.put('V', 5);
lookup.put('X', 10);
lookup.put('L', 50);
lookup.put('C', 100);
lookup.put('D', 500);
lookup.put('M', 1000);

        
        for (char letter : a){
            int letterAsNum = lookup.get(letter);
            
            if (prev < letterAsNum){      int number = (prev - letterAsNum)*(-1);RunningSum = RunningSum + number - prev;   }
            else { RunningSum = RunningSum + letterAsNum;}
            prev = letterAsNum; //update previous
            
        }
        
     return RunningSum;   
    }
}